### PR TITLE
Glacier delete

### PR DESCRIPTION
**Pull request by thisisshi**
Adds delete action to non-empty vaults, addresses: #1669

### CODE REVIEW

**Comment by kapilt** [comment]
thanks, i'm a little skeptical though of having the delete without handling the actual contents. requiring a non empty archive for the delete, makes me wonder at the utility of the delete. its also at odds with how we handle say s3 deletes.  note you don't need to do an sns to get the inventory, and also note that that the inventory is typically a trailing 24hr, so no guarantees there either to get to non empty.

**Comment by thisisshi** [comment]
Thanks Kapil, Yeah, I noticed that about the `initiate-job` call as well. 

With the S3 delete action, does custodian just wait indefinitely for the bucket to clear out? Vaults/Buckets with a ton of data can take a lot of time to empty out, should we include an option to `remove-archives` similar to s3's `remove-contents`? 

Additionally, since inventory is a trailing 24h, should this action update the vault policy to prevent additional uploads in the mean time and be used in a mark-sweep as a best practice?

**Comment by kapilt** [comment]
given it can't be reliably done except by a long running, multi-day phased script, and it can run against vaults with very large object counts, I don't think its something that we should do as a custodian action. I'd be happy to have it as a script in tools, but we've generally tried to back off data plane resources when dealing with large cardinalities, and push them out to other dedicated tools (c7n_salactus for s3 objects, etc) rather then put them in policy language where they might cause folks unintended headaches.

File: c7n/resources/glacier.py [hunk 1]
```diff
@@ -162,3 +162,49 @@ def process_resource(self, client, resource):
...
+    schema = type_schema('delete')
+    permissions = ('glacier:DeleteVault', 'glacier:AbortMultipartUpload')
+
+    def process(self, resources):
+        results = []
+        client = local_session(self.manager.session_factory).client('glacier')
+
+        for r in resources:
+            try:
+                results += filter(None, [self.process_resource(client, r)])
```
Comment (jhnlsn) [inline c7n/resources/glacier.py]: Can you iterate over the resources using threading?

**Comment by thisisshi** [comment]
Closing for now, possibly revisit delete via a tool like c7n_salactus

**Pull request closed without merging**